### DISCUSSION TITLE
Consiglio per Hub HA

### DISCUSSION THREAD

**Discussion by gup133** [indomus/forum#2881] [Hardware & Firmware]
Salve a tutti, sono un nuovo utente, neofita di HA (anche se sto cercando di imparare in fretta) e di programmazione linux, ma che ha letteralmente perso la testa per questo OS, volendo realizzare un impianto domotico personalizzato.
Da fine ottobre ho cominciato a documentarmi un pò per realizzare un piccolo impianto che mi consentisse di automatizzare e di "controllare" da remoto qualche aspetto della mia abitazione.
Ho quindi installato HA sul mio nas QNAP (che uso prevalentemente come server PLEX) attraverso VM Ware e ho integrato, nell'ordine:
1. Termostato CUBE S-Net per caldaia Ariston
2. Impianto di Allarme Somfy Home Alarm Essential (attraverso MQTT)
3. Condizionatori Daikin
4. Tapparelle elettriche attraverso shelly 2.5PM
5. Luci di tutta casa attraverso shelly 2.5 PM
6. Videocamera da interno EZVIZ C6 (al momento no motioneye ma conto di installarlo a breve)
7. Videocamera da esterno EZVIZ C3W Pro (al momento no motioneye ma conto di installarlo a breve)
8. Pannello Sonoff NSPanel Pro che ho modificato per utilizzarlo come pannello centrale di controllo di HA

Da neofita ho cercato di documentarmi più che potevo per capire come integrare le periferiche in HA e soprattutto come renderle funzionali alle mie esigenze. Ovviamente, man manco che le conoscenze crescono, aumentano bisogni e, soprattutto esigenze.
Il prossimo passo sarà integrare Alexa in HA (attraverso il metodo gratuito e un pochino più complicato spiegato alla perfezione in indomus) per poter integrare i comandi vocali e, soprattutto, la possibilità di armare/disarmare l'allarme vocalmente dal momento che Alexa non lo riconosce (mentre HA si).
Dopodiche cercherò di imparare come si creano le automazioni/routine.

Ora, dopo tutto questo preambolo, vorrei affrontare il discorso Hardware per far girare HA: al momento, infatti, come vi dicevo, utilizzo un Nas QNAP (cpu celeron), assolutamente perfetto e molto veloce; tuttavia, i consumi del NAS non sono economicissimi e dovendolo lasciare accesso h24 volevo capire se esiste un'alternativa valida e un pò meno esosa in termini di consumi. Il discorso, per farla breve, è capire se per quelle che sono le mie esigenze (di cui sopra), un raspberry Pi (4 o 5) può fare al caso mio, oppure no.
Avevo anche visto il progetto Home Assistant Yellow Kit, molto carino, ma decisamente spropositato in termini di costo.

Cosa ne pensate? Mi consigliate il raspberry?

**Comment by indomus** [comment]
Pronti: https://indomus.it/formazione/domotica-su-home-assistant-mi-conviene-un-nuc-un-raspberry-pi-o-altro/

**Reply by gup133** [reply]
Ciao! ho letto questa guida, ma proprio da questa lettura mi è sorto il dubbio in merito alla validità (o meno) del Raspberry Pi per le mie esigenze....in altre parole, pensate che sia sufficiente a gestire le periferiche del mio impianto, o rischio di essere limitato?

**Reply by indomus** [reply]
Azz andiamo male, doveva assolutamente levarteli i dubbi, anziché mettertene di nuovi :D
Raspberry Pi, ora come ora, non vale assolutamente la pena - se non un modello 4 preso davvero iper scontato (50-60 euro in kit completo). Non vale quello che costa e non ti offre le prestazioni di cui puoi e potrai necessitare.
Optare sicuramente su [un Mini PC](https://indomus.it/componenti/mini-pc-intel-nuc-e-similari/).

**Reply by gup133** [reply]
Il problema del Mini PC è in rapporto a ciò che uso al momento, ovvero il NAS: consuma più un QNAP o un Mini PC?
L'obiettivo sarebbe abbassare i consumi....

**Reply by KC-inDomus** [reply]
beh, qnap sulla carta consuma di più non foss'altro perché deve gestire anche i dischi

**Reply by gup133** [reply]
Su questo hai ragione, vorrei capire quanto sia la differenza in termini di consumo elettrico, posto che il mini pc dovrebbe in ogni caso essere acceso H24. Questo ovviamente per capire se vale la pena partendo dall'assunto che il QNAP già ce l'ho, mentre il mini pc dovrei acquistarlo (e quindi ci sarebbe un delta iniziale di investimento di almeno un paio di centinaia di euro)...

**Reply by snakuzzo** [reply]
Sulle recensioni dei mini PC, se non erro, c'è scritto.
Però il discorso è...che ci devi fare? Se cominci a stressare la cpu ad esempio con transcodifiche per via delle telecamere, il consumo ovviamente aumenta. Di base, almeno sul mio PC, siamo sui 7-8 W di assorbimento, se non ricordo male.
Per il QNAP puoi verificarlo direttamente tu con una presa smart (sperando sia tarata correttamente)

**Comment by gup133** [comment]
> Sulle recensioni dei mini PC, se non erro, c'è scritto. Però il discorso è...che ci devi fare? Se cominci a stressare la cpu ad esempio con transcodifiche per via delle telecamere, il consumo ovviamente aumenta. Di base, almeno sul mio PC, siamo sui 7-8 W di assorbimento, se non ricordo male. Per il QNAP puoi verificarlo direttamente tu con una presa smart (sperando sia tarata correttamente)

Ti ringrazio molto, farò la prova con una presa smart in modo da esser certo del consumo del QNAP per poi fare un paragone con il Mini PC così da avere un termine di paragone.
In ogni caso, riguardo le mie esigenze, il mio HA dovrebbe controllare solamente gli shelly delle luci/tapparelle elettriche, accensione/spegnimeno condizionatori e caldaia, armare/disarmare allarme con annesse due videocamere (una interna e una esterna sul terrazzo). A livello di transcodifica praticamente nulla, nella plancia principale ho inserito le rtsp delle due videocamere ma in realtà potrei disabilitarle tranquillamente se queste dovessere stressare troppo il mini pc.

**Reply by snakuzzo** [reply]
Che sia rtsp, non vuol dire nulla. Anche con RTSP potrebbe partire la transcodifica. Dipende anche dal client che utilizzi per visualizzarle (app, browser, altro)...cambia praticamente tutto.
Poi dipende se il flusso è continuo oppure no e se prevedi transcodifica sempre oppure solo quando visualizzi. 
Anche solo con due cam potresti avere un carico eccessivo di CPU.
Su questo tema c'è dietro un mondo, e l'unico modo per valutare effettivamente le conseguenze...è provare.
Il resto che hai menzionato invece è praticamente a "consumo quasi zero".

**Comment by michele-diurno** [comment]
Come hai risolto? Sei rimasto con il Nas?

**Reply by gup133** [reply]
Ciao Michele, al momento si, vorrei prima valutare per bene il funzionamento per poi decidere!

**Comment by michele-diurno** [comment]
Ti racconto la mia esperienza. Ho installato tutto sul synology in quanto è sempre accesso perchè lo utilizzo come media player dentro e fuori casa, stoccaggio di files come foto, documenti ecc. ecc. Non  vedovo motivo di spendere altri soldi per un rpi che sarebbe stato cmq un ulteriore dispositivo acceso 24 su 24. Posso dirti che su macchina virtuale funziona benissimo!!! Non ti accorgerai della differenza in bolletta alla fine i nas sono fatti per lavorare 24 su 24 quindi consumano il giusto, poi dipende anche dagli HDD che vai a montare. Per quanto riguarda HA può semplificare la vita certo non la stravolge ma può tornare utile. Ovvio che bisogna investire sui dispositivi per domotizzare la casa ma ognuno da priorità a quello che serve poi con il tempo si può pensare di aggiungere delle chicche che possono dare quel qualcosina in più anche a livello estetico più che funzionale. E' cmq un modo per mettersi alla prova come un vero hobby, tempo permettendo.
